extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -388,8 +389,18 @@ export class JupyterServerInstallation {
 	/**
 	 * Prompts user to upgrade certain python packages if they're below the minimum expected version.
 	 */
-	public promptForPackageUpgrade(): Promise<void> {
-		return this.upgradePythonPackages(true, false);
+	public async promptForPackageUpgrade(): Promise<void> {
+		if (this._installInProgress) {
+			this.apiWrapper.showInfoMessage(msgSkippedPackageUpgrade);
```

Comment:
So one thing with this is that currently you'd show this message every time - regardless of whether there were packages to actually update or not. It'd be slightly better to either A. only show the message if we were supposed to do an upgrade but couldn't or B. Update the warning message to just start with `Skipping notebook package upgrade check ...`

Reply:
Currently changing the install logic so that all Python notebooks wait for an install to finish before starting. This message can then be removed.